### PR TITLE
Build output trimming.

### PR DESCRIPTION
**Pull request by edespino**
This PR turns down the volume of build output across multiple compilation jobs to help relieve the weight on the Concourse UI. There is no functional difference in the build process.

### CODE REVIEW

File: concourse/scripts/compile_gpdb.bash [hunk 1]
```diff
@@ -102,7 +102,7 @@ function build_gppkg() {
 function unittest_check_gpdb() {
   pushd $GPDB_SRC_PATH
     source $GREENPLUM_INSTALL_DIR/greenplum_path.sh
-    make GPROOT=/usr/local -s unittest-check
+    make GPROOT=/usr/local -s unittest-check | grep -v "\[ RUN"
```
Comment (d) [inline concourse/scripts/compile_gpdb.bash]: Wait really?

File: concourse/scripts/compile_gpdb.bash [hunk 2]
```diff
@@ -112,15 +112,15 @@ function build_pxf() {
   export BUILD_NUMBER="${TARGET_OS}"
   export PXF_HOME="${GREENPLUM_INSTALL_DIR}/pxf"
   export JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8
-  make install -s DATABASE=gpdb
+  make install -s DATABASE=gpdb | grep -v "Download http"
```
Comment (d) [inline concourse/scripts/compile_gpdb.bash]: If this is noise, why don't we change the PXF builds files (Makefile and maven files) to reduce noise there?

Comment (edespino) [reply concourse/scripts/compile_gpdb.bash]: I've hunted and hunted for the proper way to turn down the Download noise volume. Generally, this is not my preferred option. I'll take a look again to see if I can do just that. Thanks for looking at this.

**Comment by d** [comment]
This seems to be a solution with a clear problem in mind. If so, do we have any metrics that we measure against which we can show demonstrable improvement?

Pivots working on Greenplum are feeling the pain and it is our responsibility to fix it, so I really appreciate you taking a crack at this, Ed. My concern is taking a hammer at whatever looks like a nail.

**Comment by edespino** [comment]
@d I had considered identifying how much this PR reduced output wise. I wish it was easier to get logs out of our CI system for a specific job. Let me see what I can do to identify the exact savings in output.  Thanks for your feedback!

File: concourse/scripts/compile_gpdb.bash (hunk 1 above)
Comment (edespino) [reply concourse/scripts/compile_gpdb.bash]: As an option, is there a way to provide a summary of the entire cmockery run?

Comment (jimmyyih) [reply concourse/scripts/compile_gpdb.bash]: This change looks fine to me.  It's just removing the RUN line but you still get the OK or FAIL line.

Comment (edespino) [reply concourse/scripts/compile_gpdb.bash]: That is correct @jimmyyih .. the actual test run status remains.

Comment (asimrp) [reply concourse/scripts/compile_gpdb.bash]: Alternatively, `cmockery.c` can be modified to not print the `[ RUN ]` line.  The `grep` at the end removes colors introduced by cmockery.  I don't see us upgrading cmockery so chancing this third-party sourcecode should be fine.

**Comment by jchampio** [comment]
As nervous as it makes me to grep out parts of the test output -- I've been actively pursuing problems recently where the Concourse output is obscuring bugs -- I understand there's a real problem we have to solve here...

Are the Concourse maintainers aware that we're having to tear out output just to get the UI to function nicely?

**Comment by edespino** [comment]
@jchampio - Here is a list of various Concourse UI performance issues that we encounter regularly. This PR is attempts to address spurious output scattered about. It is not intended to do away with useful information.

- [Concourse freezes browser for minutes at a time when opening (large) build log #1917](https://github.com/concourse/concourse/issues/1917)
- [Lengthy task output renders extremely slowly #1912](https://github.com/concourse/concourse/issues/1912
- [The build numbers on the build page should do lazy loading and have a "more" button #1855](https://github.com/concourse/concourse/issues/1855)
- [Poor performance #1843](https://github.com/concourse/concourse/issues/1843)
- [Build view taking much longer to load #1771](https://github.com/concourse/concourse/issues/1771)
- [build page loading is very slow #1543](https://github.com/concourse/concourse/issues/1543)
- [96 jobs make browsers unusably slow #991](https://github.com/concourse/concourse/issues/991)

**Comment by hsyuan** [comment]
@edespino I have seen a lot of download output from pxf. Why don't they just embed these packages into the image or s3 as a single tarball all together? Not only it will make it less noisy, but it will also save a lot of time.

File: concourse/scripts/compile_gpdb.bash (hunk 1 above)
Comment (edespino) [reply concourse/scripts/compile_gpdb.bash]: @asimrp - As you suggested, I have updated cmockery.c to not print the ```[RUN]``` lines. I have removed the previous change.

File: src/test/unit/cmockery/cmockery.c [hunk 3]
```diff
@@ -1770,9 +1770,6 @@ int _run_test(
 #endif // !_WIN32
 	}
 
-	if (function_type == UNIT_TEST_FUNCTION_TYPE_TEST) {
```
Comment (asimrp) [inline src/test/unit/cmockery/cmockery.c]: Nice!

**Comment by edespino** [comment]
Dev branch for this PR has been rebased to master (456f57e6c1). The `gpdb_master` and `gpdb_master_without_asserts` pipelines are in green state.

**PR pipeline passed successfully.**

**Dev pipeline passed successfully.** For future reference, here is the basis for setting the dev pipeline used for this PR.

Dev Pipeline definition:

```
## Generated by gen_pipeline.py at: 2017-12-22 12:18:15.388172
## Template file: gpdb-tpl.yml
## OS Types: ['centos6', 'centos7', 'sles', 'aix7', 'win', 'ubuntu16']
## Test Sections: ['ICW']
```

Dev Pipeline setting definition:

```
fly -t gpdb-dev \
    set-pipeline \
    -p gpdb-builds-espino \
    -c gpdb-builds-espino.yml \
    -l <common creds> \
    -l <master creds> \
    -v tf-bucket-path=dev/builds/ \
    -v bucket-name=gpdb5-concourse-builds-dev \
    -v gpdb-git-branch=build-trimming \
    -v gpdb-git-remote=https://github.com/edespino/gpdb.git
```

**Pull request merged**